piptools/scripts/compile.py

```diff
@@ -224,6 +224,14 @@ def _get_default_option(option_name: str) -> Any:
 @click.option(
     "--pip-args", "pip_args_str", help="Arguments to pass directly to the pip command."
 )
+@click.option(
+    "--resolver",
+    "resolver_name",
+    type=click.Choice(("legacy", "2020")),
```

Comment:
I don't think it's very user-friendly to call a resolver `"2020"`. It refers to a pip version which is an implementation detail. Maybe call it `"backtracking"`? Or maybe call it `resolvelib` (but that's also too internal).

Reply:
Backtracking is okay. Addressed in c5fdc7e.